xplique/metrics/fidelity.py

```diff
@@ -436,9 +484,9 @@ def evaluate(self,
 
         baselines_flatten = baselines.reshape(self.inputs_flatten.shape)
 
-        steps = np.linspace(1, self.max_nb_perturbation, self.steps, dtype=np.int32)
+        steps = np.linspace(0, self.nb_features * self.max_percentage_perturbed, self.steps+1,
```

Comment:
Seeing this, I think it is better to define self.max_nb_pertubations as it was previously made!